java/org/apache/tomcat/util/scan/StandardJarScanner.java

```diff
@@ -219,7 +284,7 @@ public void scan(JarScanType scanType, ServletContext context,
                     if (isWebapp) {
                         isWebapp = isWebappClassLoader(classLoader);
                     }
-                    URL[] urls = ((URLClassLoader) classLoader).getURLs();
+                    URL[] urls = getURLs((URLClassLoader) classLoader);
```

Comment:
JARs loaded by the web application class loader will never load JARs via the Manifest. It is a waste of time checking that class loader.
You should also check if the common, shared and catalina class loaders check the Manifest. I suspect they don't.